qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,342 @@
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    @staticmethod
+    def _log_gradients(
+        iteration: int,
+        cur_grads: Tuple[float, PauliSumOp],
+        max_grad: Tuple[float, PauliSumOp | None],
+    ):
+        if logger.isEnabledFor(logging.INFO):
+            gradlog = f"\nGradients in iteration #{str(iteration)} "
+            gradlog += "\nID: Excitation Operator: Gradient  <(*) maximum>"
+            for i, grad in enumerate(cur_grads):
+                gradlog += f"\n{str(i)}: {str(grad[1])}: {str(grad[0])}"
+                if grad[1] == max_grad[1]:
+                    gradlog += "\t(*)"
+            logger.info(gradlog)
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: OperatorBase,
+        aux_operators: Optional[ListOrDict[PauliSumOp]] = None,
+    ) -> AdaptVQEResult:
+        """Computes the ground state.
+
+        Args:
+            operator: Operator to evaluate
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: If a solver other than VQE
+                (`qiskit.algorithms.minimum_eigen_solvers.vqe.py`) or a ansatz other than
+                EvolvedOperatorAnsatz (`qiskit.circuit.library.evolved_operator_ansatz.py`)
+                is provided or if the algorithm finishes due to an unforeseen reason.
+            ValueError: If the grouped property object returned by the driver does not contain a
+                main property as requested by the problem being solved (`problem.main_property_name`)
+            QiskitError: If the user-provided `aux_operators` contain a name which clashes
+                with an internally constructed auxiliary operator. Note: the names used for the
+                internal auxiliary operators correspond to the `Property.name` attributes which
+                generated the respective operators.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An AdaptVQEResult (`qiskit.algorithms.minimum_eigen_solvers.adapt_vqe.AdaptVQEResult`)
+            which is a VQEResult (`qiskit.algorithms.minimum_eigen_solvers.vqe.VQEResult`) but also
+            includes runtime information about the AdaptVQE algorithm like the number of iterations,
+            finishing criterion, and the final maximum gradient.
+        """
+        if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
+            raise QiskitError(
+                "The AdaptVQE algorithm requires the use of the evolved operator ansatz"
+            )
+        # We construct the ansatz once to be able to extract the full set of excitation operators.
+        self._tmp_ansatz._build()
+
+        prev_op_indices: List[int] = []
+        theta: List[float] = []
+        max_grad: Tuple[float, Optional[PauliSumOp]] = (0.0, None)
```

Comment:
I noticed in another setting, that we should reset `self._excitation_list` here, in order to allow re-usability of the algorithm:
```suggestion
        max_grad: Tuple[float, Optional[PauliSumOp]] = (0.0, None)
        self._excitation_list = []
```